girder/utility/plugin_utilities.py

```diff
@@ -327,10 +325,49 @@ def findAllPlugins(curConfig=None):
             'version': '',
             'dependencies': set()
         }
-        allPlugins[entry_point.name].update(
-            getattr(entry_point.load(), 'config', {})
-        )
+        configJson = os.path.join('girder', 'plugin.json')
+        configYml = os.path.join('girder', 'plugin.yml')
+        data = {}
+        try:
+            if pkg_resources.resource_exists(entry_point.name, configJson):
+                with pkg_resources.resource_stream(
+                        entry_point.name, configJson) as conf:
+                    try:
+                        data = json.load(codecs.getreader('utf8')(conf))
+                    except ValueError as e:
+                        print(
+                            TerminalColor.error(
+                                'ERROR: Plugin "%s": plugin.json is not valid '
+                                'JSON.' % entry_point.name))
+                        print(e)
+            elif pkg_resources.resource_exists(entry_point.name, configYml):
+                with pkg_resources.resource_stream(
+                        entry_point.name, configYml) as conf:
+                    try:
+                        data = yaml.safe_load(conf)
+                    except yaml.YAMLError as e:
+                        print(
+                            TerminalColor.error(
+                                'ERROR: Plugin "%s": plugin.yml is not valid '
+                                'YAML.' % entry_point.name))
+                        print(e)
+        except ImportError:
```

Comment:
`pkg_resources.resource_exists` can raise it.  Specifically, it imports the package via __import__.

If the package tries to import something that is unavailable the import error is thrown.  It will probably also be thrown by entry_point.load(), but the error message could be different, and, if I remember, it seemed more lucid.